### ISSUE TITLE
Is it possible to sequentialy add up gradients to increase effective batch size?

### ISSUE THREAD

**Issue by marcel1991** [torch/nn#1244]
I was wondering if it is possible to add up the gradients that were computed over multiple batches. My problem is that I run out of memory and therefore can't increase the batch size on my gpu anymore. Therefore I thought that I could sequentially compute the gradients for multiple mini batches and just add them up after n iterations to get an effective batchSize = n x bachtSize. This would be quite usefull to improve the learning at the end of the whole training.

So I think to accomplish this I just need to control when the gradient adding is zeroed. Is zeroGradParameters() the function that does this? And another general question: does accGradParameters() add the gradients in a module for every sample of the batch or what does it accumulate? I read the docs but this is still not clear to me.

**Comment by iamalbert** [comment]
yes, you can do `model:backward()` for many times, followed by a `model:updateParameters(lr)` call to update the parameters

**Comment by marcel1991** [comment]
Hm okay I gonna try that out later. And can you also tell me what is accumulated in accGradParameters()? Is it the gradient of the different samples in the minibatch?
